src/Microsoft.Framework.FileSystemGlobbing/Matcher.cs

```diff
@@ -1,27 +1,28 @@
-// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System.Collections.Generic;
 using Microsoft.Framework.FileSystemGlobbing.Abstractions;
-using Microsoft.Framework.FileSystemGlobbing.Infrastructure;
+using Microsoft.Framework.FileSystemGlobbing.Internal;
+using Microsoft.Framework.FileSystemGlobbing.Internal.Patterns;
 
 namespace Microsoft.Framework.FileSystemGlobbing
 {
     public class Matcher
     {
-        public IList<Pattern> IncludePatterns { get; } = new List<Pattern>();
+        internal IList<IPattern> IncludePatterns { get; } = new List<IPattern>();
```

Comment:
They don't need to be properties at all. I'll cut the dependency from `MatcherContext` to `Matcher`